File: frontend/src/types/restaurant.ts
/**
 * Restaurant types matching backend Pydantic schemas.
 */

export interface Review {
  summary: string;
  recommended_dishes: string[];
  source_url?: string;
  source_type?: string;
  sentiment_score?: number;
}

export interface Restaurant {
  id: string;
  name: string;
  slug?: string;
  address: string;
  price_tier: 1 | 2 | 3 | 4;
  cuisine_tags: string[];
  review?: Review;
  latitude: number;
  longitude: number;
  google_maps_url?: string;
  vibe?: string;
  
  // Scores
  buzz_score: number;
  sentiment_score: number;
  total_mentions: number;
  
  // Flags
  is_trending: boolean;
}

export interface SearchResponse {
  results: Restaurant[];
  total: number;
  query?: string;
  filters?: SearchFilters;
}

export interface SearchFilters {
  price_min?: number;
  price_max?: number;
  cuisine?: string[];
  sort_by?: string;
  sort_order?: string;
}

export interface SearchParams {
  q?: string;
  price_min?: number;
  price_max?: number;
  cuisine?: string[];
  sort_by?: 'buzz_score' | 'sentiment_score' | 'total_mentions';
  sort_order?: 'asc' | 'desc';
  limit?: number;
}

File: backend/models/restaurant.py
from typing import List, Optional
from pydantic import BaseModel, Field

class Restaurant(BaseModel):
    """Core Identity with Google Identifiers."""
    id: Optional[str] = None
    name: str
    slug: Optional[str] = None
    address: str
    city: str = "Toronto"
    latitude: float = 0.0
    longitude: float = 0.0
    price_tier: int = Field(2, ge=1, le=4)
    photo_url: Optional[str] = None
    vibe: Optional[str] = None
    
    # Essential Google Info
    google_place_id: Optional[str] = None
    google_maps_url: Optional[str] = None
    
    embedding: Optional[List[float]] = None